src/debugger/nodeDebugWrapper.ts

```diff
@@ -135,8 +138,41 @@ Telemetry.init("react-native-debug-adapter", version, true).then(() => {
             args.target || "simulator"
         ];
 
+        if (!isNullOrUndefined(args.logCatArguments)) { // We add the parameter if it's defined (adapter crashes otherwise)
+            args.args = args.args.concat([parseLogCatArguments(args.logCatArguments)]);
+        }
+
         originalNodeDebugSessionLaunchRequest.call(this, request, args);
     };
 
+    // Intecept the "launchRequest" instance method of NodeDebugSession to interpret arguments
+    const originalNodeDebugSessionDisconnectRequest = nodeDebug.NodeDebugSession.prototype.disconnectRequest;
+    function customDisconnectRequest(response: any, args: any): void {
+        try {
+            // First we tell the extension to stop monitoring the logcat, and then we disconnect the debugging session
+            const extensionMessageSender = new ExtensionMessageSender();
+            extensionMessageSender.sendMessage(ExtensionMessage.STOP_MONITORING_LOGCAT)
+                .finally(() => originalNodeDebugSessionDisconnectRequest.call(this, response, args))
+                .done(() => {}, reason => // We just print a warning if something fails
+                    process.stderr.write(`WARNING: Couldn't stop monitoring logcat: ${reason.message || reason}\n`));
+        } catch (exception) {
+            // This is a "nice to have" feature, so we just fire the message and forget. We don't event handle
+            // errors in the response promise
+            process.stderr.write(`WARNING: Couldn't stop monitoring logcat. Sync exception: ${exception.message || exception}\n`);
+            originalNodeDebugSessionDisconnectRequest.call(this, response, args);
+        }
+    }
+    nodeDebug.NodeDebugSession.prototype.disconnectRequest = customDisconnectRequest;
+
     vscodeDebugAdapterPackage.DebugSession.run(nodeDebug.NodeDebugSession);
-});
\ No newline at end of file
+});
+
+function parseLogCatArguments(userProvidedLogCatArguments: any) {

```

Comment:
Is there a reason you have these down the bottom rather than toward the top? The code above this is executed immediately, and I find that having function definitions afterwards makes that harder to notice.